components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -782,9 +786,36 @@ private String getSubjectClaimFromUserStore(String subjectClaimUri, Authenticate
         AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) IdentityTenantUtil
                 .getRealm(authenticatedUser.getTenantDomain(), authenticatedUser.toFullQualifiedUsername())
                 .getUserStoreManager();
+        if (userStoreManager.isExistingUserWithID(authenticatedUser.getUserId())) {
+            return userStoreManager
+                    .getUserClaimValueWithID(authenticatedUser.getUserId(), subjectClaimUri, null);
+        }
 
-        return userStoreManager
-                .getUserClaimValueWithID(authenticatedUser.getUserId(), subjectClaimUri, null);
+        /*
+            If the user is not in the user store, check if the user can be resolved from ancestor organizations and
+            resolve the subject claim.
+        */
+        try {
+            String organizationId = OAuth2ServiceComponentHolder.getInstance().getOrganizationManager()
+                    .resolveOrganizationId(authenticatedUser.getTenantDomain());
+            Optional<String> userResideOrgId = OAuth2ServiceComponentHolder.getOrganizationUserResidentResolverService()
+                    .resolveResidentOrganization(authenticatedUser.getUserId(), organizationId);
+            if (!userResideOrgId.isPresent()) {
+                throw new org.wso2.carbon.user.core.UserStoreException("Could not resolve the user reside " +
```

Comment:
```suggestion
                throw new org.wso2.carbon.user.core.UserStoreException("Could not resolve the user resident " +
```

Reply:
This part were removed by this commit - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2103/commits/b3a1b2a1838130a38dc197f0ab9658d22237ab37